File: 0283-MoveZeroes/cpp/MoveZeroes.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int insertPos = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) {
                nums[insertPos++] = nums[i];
            }
        }
       
        while (insertPos < nums.size()) {
            nums[insertPos++] = 0;
        }
    }
};

int main() {
    Solution solution;
    vector<int> nums = { 0, 1, 0, 3, 12 };

    cout << "Before moveZeroes: ";
    for (int num : nums) {
        cout << num << " ";
    }
    cout << endl;

    solution.moveZeroes(nums);

    cout << "After moveZeroes:  ";
    for (int num : nums) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: 0283-MoveZeroes/python/MoveZeroes.py
from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        insert_pos = 0
        
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[insert_pos] = nums[i]
                insert_pos += 1
        
        while insert_pos < len(nums):
            nums[insert_pos] = 0
            insert_pos += 1


nums = [0, 1, 0, 3, 12]
print("Before moveZeroes:", nums)

solution = Solution()
solution.moveZeroes(nums)

print("After moveZeroes: ", nums)
